app/controllers/lti_nrps_controller.rb

```diff
@@ -84,11 +86,70 @@ def send_nrps_request
     lcd.last_synced = DateTime.current
     lcd.save
 
-    render json: members.as_json
+    # Update the roster with the retrieved set of members
+    @cuds = parse_members_data(lcd, members.as_json)
+    @sorted_cuds = @cuds.sort_by { |cud| cud[:color] || "z" }.reverse
+    @course = lcd.course
   end
 
 private
 
+  def parse_members_data(lcd, members_data)
+    cuds = CourseUserDatum.where(course_id: lcd.course_id, instructor: false,
+                                 course_assistant: false).to_set
+    email_to_cud = {}
+    cuds.each do |cud|
+      email_to_cud[cud.user.email] = cud
+    end
+
+    cud_view = []
+    members_data.each do |user_data|
+      next unless user_data["roles"].include? "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner"
```

Comment:
Can you also check if "status":"Active" in addition to them having the learner role? Seems like canvas just removes them from the query so this works but for other platforms it may be different

Reply:
Added